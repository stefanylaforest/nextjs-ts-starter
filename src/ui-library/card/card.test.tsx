import React from 'react';
import { render } from '@testing-library/react';
import Card from './card';

describe('Card Component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Card
        header={<h3>Hello World</h3>}
        body={<img src="images/card-photo-example-1.jpg" alt="" />}
        footer={'read more'}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it('should have a footer', () => {
    const { container } = render(<Card footer={'view'} />);
    expect(container).toMatchSnapshot();
  });
});
