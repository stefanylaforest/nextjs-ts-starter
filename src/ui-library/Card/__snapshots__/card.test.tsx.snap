// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should have a footer 1`] = `
<div>
  <div
    class="cardWrapper"
  >
    <div
      class="cardFooter"
    >
      view
    </div>
  </div>
</div>
`;

exports[`Card Component should match snapshot 1`] = `
<div>
  <div
    class="cardWrapper"
  >
    <div
      class="cardHeader"
    >
      <h3>
        Hello World
      </h3>
    </div>
    <div
      class="cardBody"
    >
      <img
        alt=""
        src="images/card-photo-example-1.jpg"
      />
    </div>
    <div
      class="cardFooter"
    >
      read more
    </div>
  </div>
</div>
`;
